react/features/subtitles/middleware.ts

```diff
@@ -134,18 +135,34 @@ function _endpointMessageReceived(store: IStore, next: Function, action: AnyActi
         name
     };
 
+    let newTranscriptMessage: ITranscriptMessage | undefined;
+
     if (json.type === JSON_TYPE_TRANSLATION_RESULT && json.language === language) {
         // Displays final results in the target language if translation is
         // enabled.
 
-        const newTranscriptMessage = {
+        newTranscriptMessage = {
             clearTimeOut: undefined,
-            final: json.text,
+            final: json.text?.trim(),
             participant
         };
 
-        _setClearerOnTranscriptMessage(dispatch, transcriptMessageID, newTranscriptMessage);
-        dispatch(updateTranscriptMessage(transcriptMessageID, newTranscriptMessage));
+        const cachedTranscriptMessage = state['features/subtitles']._cachedTranscriptMessages?.get(transcriptMessageID);
+
+        if (cachedTranscriptMessage) {
+            const cachedText = (cachedTranscriptMessage.stable || cachedTranscriptMessage.unstable)?.trim();
```

Comment:
Do we need to trim here again? We are already timming on assingment above?

Reply:
The assignment is for the new message and this is for the cached message.